examples/read_eeg_from_bids.py

```diff
@@ -0,0 +1,165 @@
+# Let's see whether the data has been downloaded using a quick visualization
+# of the directory tree.
+
+data_dir = os.path.join(bids_root, 'raw')
+print_dir_tree(data_dir)
+
+###############################################################################
+# Step 2: Formatting as BIDS
+# --------------------------
+#
+# Let's start by formatting a single subject. We are reading the data using
+# MNE-Python's io module and the :func:`read_raw_edf` function. Note that we
+# must use `preload=False`, the default in MNE-Python. It prevents the data
+# from being loaded and modified when converting to BIDS.
+#
+# Note that kind and acquisition currently almost stand for the same thing.
+# Please read the BIDS docs to get acquainted.
+for edf_fpath in edf_fpaths:
+    # create the BIDS directory structure
+    if not os.path.exists(bids_root):
+        print("Making bids root directory.")
+        make_bids_folders(
+            output_path=bids_root,
+            session=test_sessionid,
+            subject=test_subjectid,
+            kind=modality,
+        )
+
+    ''' Write data file into BIDS format '''
+    # add a bids run
+    bids_basename = make_bids_basename(
+        subject=test_subjectid,
+        session=test_sessionid,
+        run=test_runid,
+        acquisition=modality,
+    )
+    print("Loading filepath: ", edf_fpath)
+    print("Writing to bidsroot: ", bids_root)
+    print("Bids basenmae; ", bids_basename)
+    # call bidsbuilder pipeline
+    bids_root = BidsConverter.convert_to_bids(
+        edf_fpath=edf_fpath,
+        bids_root=bids_root,
+        bids_basename=bids_basename,
+        acquisition=modality,
+        line_freq=line_freq,
+        overwrite=True,
+    )
+###############################################################################
+# What does our fresh BIDS directory look like?
+print_dir_tree(bids_root)
+
+###############################################################################
+# Step 3: Check and compare and read in the data
+# ------------------------------------------------------------
+# Now we have written our BIDS directory.
+
+if modality in ["ecog", "seeg"]:
+    kind = "ieeg"
+elif modality == "eeg":
+    kind = "eeg"
+bids_fname = bids_basename + f"_{kind}.fif"
+
+print("Trying to read from: ", bids_fname)
+
+# use MNE-BIDS function to read in the data
+raw = read_raw_bids(bids_fname, bids_root)
+
+print("Read successfully using MNE-BIDS")
+
+# use BidsRun object, which just simply adds additional functionality
+bidsrun = BidsRun(bids_root, bids_fname)
+raw = bidsrun.load_data()
+
+print(raw)
+
+###############################################################################
+# Step 4: Run BIDS-Validate
+# ------------------------------------------------------------
+# Now we have written our BIDS directory.
+# save a fif copy and reload it
+# TODO: figure out why this isn't returning valid, when it is valid.
```

Comment:
Call all of the checks independently to figure out which one is not passing:

- is_top_level
- is_associated_data
- is_session_level
- is_subject_level
- is_phenotypic
- is_file